test/jdk/com/sun/java/swing/plaf/gtk/bug6492108.java

```diff
@@ -0,0 +1,155 @@
+        addTextComp(parent, text);
+
+        text = type.newInstance();
+        text.setEnabled(false);
+        addTextComp(parent, text);
+
+        text = type.newInstance();
+        text.setEnabled(false);
+        text.setEditable(false);
+        addTextComp(parent, text);
+    }
+
+    private static void addTextComp(Container parent, JTextComponent text) {
+        JScrollPane sp = new JScrollPane(text);
+        text.setFocusable(false); // to avoid showing the blinking caret
+        sp.setPreferredSize(new Dimension(150, 150));
+        sp.setHorizontalScrollBarPolicy(JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+        sp.setVerticalScrollBarPolicy(JScrollPane.VERTICAL_SCROLLBAR_NEVER);
+        parent.add(sp);
+    }
+
+    protected Component createContentPane() {
+        panel = new JPanel();
+        panel.setLayout(new GridLayout(3, 4));
+        try {
+            addTextComps(panel, JTextArea.class);
+            addTextComps(panel, JEditorPane.class);
+            addTextComps(panel, JTextPane.class);
+        } catch (Throwable t) {
+            fail("Problem creating text components");
+        }
+        return panel;
+    }
+
+    private void onEDT10() {
+        requestAndWaitForFocus(panel);
+    }
+
+    private void onBackgroundThread20() {
+        // For each component on the top row, compare against the two
+        // components below in the same column.  All three components in
+        // that column should be the same pixel-for-pixel.
+        for (int count = 0; count < 4; count++) {
+            Component ref = panel.getComponent(count);
+            Point loc = ref.getLocationOnScreen();
+            Rectangle refRect =
+                new Rectangle(loc.x, loc.y, ref.getWidth(), ref.getHeight());
+            BufferedImage refimg = robot.createScreenCapture(refRect);
+
+            for (int k = 1; k < 3; k++) {
+                int index = count + (k*4);
+                Component test = panel.getComponent(index);
+                loc = test.getLocationOnScreen();
+                Rectangle testRect =
+                    new Rectangle(loc.x, loc.y,
+                                  test.getWidth(), test.getHeight());
+                BufferedImage testimg = robot.createScreenCapture(testRect);
+
+                if (refimg.getWidth() != testimg.getWidth()
+                   || refimg.getHeight() != testimg.getHeight())
+                {
+                    fail("Test image size must match reference image size");
+                }
+
+                for (int y = 0; y < refimg.getHeight(); y++) {
+                    for (int x = 0; x < refimg.getWidth(); x++) {
+                        int refPixel  = refimg.getRGB(x, y);
+                        int testPixel = testimg.getRGB(x, y);
+                        if (refPixel != testPixel) {
+                            fail("Image comparison failed at (" +
+                                 x + "," + y + ") for image " + index);
```

Comment:
Adding `count` and `k` to diagnostic message could help identifying the failing case quicker.

Reply:
Don't find adding `k` to diagnostic message useful.. failure message is more relevant now about the images which are compared.

Reply:
The images themselves are more useful for sure. Thanks!